Include/mydatatypes.h

```diff
@@ -0,0 +1,62 @@
+
+//  UC Berkeley - MSSE Program
+//  Chem 279-B  Introduction to Software Engineering
+//  Fall 2022
+//
+// his file contains the custom datatypes used in the Cellular Automata project
+//
+#pragma once    // Ensures that this file is only included once
+                // during compilation
+
+#include <array>
+using namespace std;
+
+enum neighborhood{VonNeumann, Moore};
+
+enum boundary{None, Periodic, Walled, CutOff};
+
+enum rule{Majority, Parity};
+
+class CellularAutomata         // The CA datastructure
+{
+    public:   
+
+        boundary b_type;     //enum code to hold boundary
+        int b_radius;       //declare a radius for the boundary
+        neighborhood n_type;     //enum code to hold neighboorhood type 
+        int num_states;         //integer code for number of states
+        
+        rule r_type;          //enum code for rule type 
+        double shortr_weight;   //double for short radius
+        double longr_weight;        //double for long radius
+        int n;              //count of cells in first dimension
+        int m;              //count of cellls in second dimension
+        int **matrix;        //pointer to 2d array for grid cells holding a state 
+
+        ///////////////////////////////////////////
+        CellularAutomata(int n, int m);              // Default constructor 
```

Comment:
We probably shouldn't initialize the matrix in the constructor since we can't return an error code. Maybe we can initialize the matrix in a new setup method. The setup method can just taking in int n and int m (`int setup(int n, int m);`. We can possibly overload the setup method to also accept the neighborhood_type, boundry_type, rule_type (`int setup(int n, int m, neighborhood neighborhood_type, boundary bound_type, int bound_radius, rule rule_type);`). We can keep the other setup methods to allow User II to modify the CA structure as they see fit after it's been setup/initialized.

Design questions (we can discuss this in person): 
- Should we use Eigen or BLAS since we are working with matrices?
- Should we allow the user to provide an matrix that already initialized and copy to the user. More advanced option would be to implement a [move](https://en.cppreference.com/w/cpp/utility/move) operation to change ownership of the matrix.

Reply:
The reason I chose to include the dimensions (n,m) in the constructor is because in order to make a dynamically allocated 2d array, I can only set the dimensions once and it cannot be changed. Before that I had included the setup_dimensions() method, so even though the 2d array was initialized in the constructor, I could change it again in the setup method by setting matrix=nullptr and initializing it again, but I chose not to do this because I don't know if it results in any memory leaks. We should discuss this in out meeting. Also as it is written this only supports 1d and 2d arrays.